### DISCUSSION TITLE
"Permission denied for schema extensions" - Using webhooks with Postgis

### DISCUSSION THREAD

**Discussion by YA9** [supabase/supabase#38607] [Questions]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I'm trying to use the supabase edge functions. I created a default test function and webhook to go along with it. The table I created the webhook on uses Postgis as a datatype (geometry) for one of the cols. When I enable a webhook trigger and add a row, I get the following error: "Permission denied for schema extensions". I found online the only workaround is to migrate the Postgis extension to Public rather than keeping it in Extensions. This is a very risky migration that I want to keep as a last resort. 

## To Reproduce

1. Create a table with a geometry col from Postgis. 
2. Create a test edge function
3. create a test webhook that triggers on insert of the table.
4. Add a row to the table (must include data in the geometry col)
5. You'll get the error.

## Expected behavior

The webhook should trigger the edge function, which it does on tables that don't use the postgis extension.

## Screenshots

<img width="580" alt="image" src="https://github.com/user-attachments/assets/a21545d7-0d6c-494f-a2ec-c8c2a269626a">

## System information

- Chrome

**Comment by YA9** [comment]
Any update? This is high priority. Thank you

**Comment by bobbybol** [comment]
We have the same issue, any report on this, or advice on what permissions to set?

**Comment by northfacejmb** [comment]
this is affecting me too. did you find a workaround?

**Comment by Kilowhisky** [comment]
I do as well

**Comment by Kilowhisky** [comment]
I figured it out. 

Webhook Triggers are run by `supabase_functions_admin`. 

So all I needed to do make it work was. 

```sql
grant all on schema extensions to supabase_functions_admin;
```

**Comment by erycson** [comment]
> grant all on schema extensions to supabase_functions_admin;

That worked for me.

**Comment by northfacejmb** [comment]
`grant all on schema extensions to supabase_functions_admin;` for some reason worked on my dev env db but not on my production one, but what did work was:
`GRANT USAGE ON SCHEMA tiger TO service_role;`

**Comment by Hallidayo** [comment]
Hi all, Thanks for your help on this one. I'm going to move over as a discussion so other users can use this for help. Thanks again!
